docs/v8-migration-api.md

```diff
@@ -77,65 +77,250 @@ Check out the [v8-migration guide](v8-migration.md) for other user-level changes
     - For `node.parameters` in Babel 7, use `node.params` in Babel 8
     - For `node.typeAnnotation` in Babel 7, use `node.returnType` in Babel 8
 
-- Rename `typeParameters` to `typeArguments` for `TSTypeQuery` ([#16679](https://github.com/babel/babel/issues/16679), [#17012](https://github.com/babel/babel/pull/17012))
+- Rename `typeParameters` to `typeArguments` for `CallExpression`, `JSXOpeningElement`, `NewExpression`, `OptionalCallExpression`, `TSInstantiationExpression`, `TSTypeQuery` and  `TSTypeReference` ([#16679](https://github.com/babel/babel/issues/16679), [#17008](https://github.com/babel/babel/pull/17008), [#17012](https://github.com/babel/babel/pull/17012), [#17020](https://github.com/babel/babel/pull/17020))
+
+  <details>
+    <summary>CallExpression</summary>
+
+    ```ts title=input.ts
+    fn<string>()
+
+    // AST in Babel 7
+    {
+      type: "CallExpression",
+      callee: Identifier("fn"),
+      arguments: [],
+      typeParameters: {
+        type: "TSTypeParameterInstantiation",
+        params: [{
+          type: "TSStringKeyword"
+        }]
+      }
+    }
 
-  ```ts title=input.ts
-  var arr: typeof Array<string>;
+    // AST in Babel 8
+    {
+      type: "CallExpression",
+      callee: Identifier("fn"),
+      arguments: [],
+      typeArguments: {
+        type: "TSTypeParameterInstantiation",
+        params: [{
+          type: "TSStringKeyword"
+        }]
+      }
+    }
+    ```
+
+  </details>
+
+  <details>
+    <summary>JSXOpeningElement</summary>
+
+    ```ts title=input.ts
+    <Component<string>/>
+
+    // AST in Babel 7
+    {
+      type: "JSXOpeningElement",
+      name: JSXIdentifier("Component"),
+      attributes: [],
+      selfClosing: true,
+      typeParameters: {
+        type: "TSTypeParameterInstantiation",
+        params: [{
+          type: "TSStringKeyword"
+        }]
+      }
+    }
 
-  // AST in Babel 7
-  {
-    type: "TSTypeQuery",
-    exprName: Identifier("Array"),
-    typeParameters: {
-      type: "TSTypeParameterInstantiation",
-      params: [{
-        type: "TSStringKeyword"
-      }]
+    // AST in Babel 8
+    {
+      type: "JSXOpeningElement",
+      name: JSXIdentifier("Component"),
+      attributes: [],
+      selfClosing: true,
+      typeArguments: {
+        type: "TSTypeParameterInstantiation",
+        params: [{
+          type: "TSStringKeyword"
+        }]
+      }
+    }
+    ```
+
+  </details>
+
+  <details>
+    <summary>NewExpression</summary>
+
+    ```ts title=input.ts
+    new Component<string>()
+
+    // AST in Babel 7
+    {
+      type: "NewExpression",
+      callee: Identifier("Component"),
+      arguments: [],
+      typeParameters: {
+        type: "TSTypeParameterInstantiation",
+        params: [{
+          type: "TSStringKeyword"
+        }]
+      }
     }
-  }
 
-  // AST in Babel 8
-  {
-    type: "TSTypeReference",
```

Comment:
Also fixed a docs error, the type here should be `TSTypeQuery`, it is overlooked during copy and paste.